commands/miner_daemon_test.go

```diff
@@ -415,3 +445,81 @@ var testConfig = &gengen.GenesisCfg{
 		},
 	},
 }
+
+func TestMinerWorker(t *testing.T) {
+	tf.IntegrationTest(t)
+
+	ctx, env := fastesting.NewTestEnvironment(context.Background(), t, fast.FilecoinOpts{})
+	defer func() {
+		require.NoError(t, env.Teardown(ctx))
+	}()
+	require.NoError(t, env.GenesisMiner.MiningStart(ctx))
+	defer func() {
+		require.NoError(t, env.GenesisMiner.MiningStop(ctx))
+	}()
+	env.RunAsyncMiner()
+
+	minerNode := env.RequireNewNodeWithFunds(1000)
+
+	t.Run("if there is no miner worker, returns error and outputs nothing", func(t *testing.T) {
+		res, err := minerNode.MinerWorker(ctx)
+		require.NotNil(t, err)
+		lastErr, err := minerNode.LastCmdStdErrStr()
+		require.NoError(t, err)
+		require.Contains(t, lastErr, "problem getting worker address")
+		require.Contains(t, lastErr, "actor not found")
+		assert.Equal(t, address.Undef, res.WorkerAddress)
+	})
+
+	t.Run("if there is a miner, shows the correct worker address", func(t *testing.T) {
+		minerAddr := requireMinerCreate(ctx, t, env, minerNode)
+
+		workerAddr, err := minerNode.MinerOwner(ctx, minerAddr)
+		require.NoError(t, err)
+
+		res, err := minerNode.MinerWorker(ctx)
+		fmt.Println(minerNode.LastCmdStdErrStr())
+		require.NoError(t, err)
+		assert.Equal(t, workerAddr.String(), res.WorkerAddress.String())
+	})
+}
+
+func TestMinerSetWorker(t *testing.T) {
+	tf.IntegrationTest(t)
+	ctx, env := fastesting.NewTestEnvironment(context.Background(), t, fast.FilecoinOpts{})
+	defer func() {
+		require.NoError(t, env.Teardown(ctx))
+	}()
+	require.NoError(t, env.GenesisMiner.MiningStart(ctx))
+	defer func() {
+		require.NoError(t, env.GenesisMiner.MiningStop(ctx))
+	}()
+	env.RunAsyncMiner()
+
+	minerNode := env.RequireNewNodeWithFunds(1000)
+
+	t.Run("fails if there is no miner worker", func(t *testing.T) {
+		res, err := minerNode.MinerWorker(ctx)
```

Comment:
Blocking: this looks like a repeat of the "if there is no miner worker..." test of the `miner worker` command above.  Shouldn't this be testing miner worker setting?

Reply:
(or just be deleted if there is no analogue in the "set" case)

Reply:
whoops! thanks for catching that. I'll update the tests.